File: files/liskov/ParentChild.cc
#include <iostream>
#include <string>

class Szulo {
public:
    void print(std::string s) {
        std::cout << s << std::endl;
    }
};

class Gyermek : public Szulo {
public:
    void echo(std::string s) {
        std::cout << s << s << std::endl;
    }
};

int main(int argc, char **argv) {
    Szulo* szulo = new Szulo();
    Szulo* szulo2 = new Gyermek();

    // Ez az, ami nem fog mukodni:
    szulo2->echo("damn, son. where'd u find dis???");

    return 0;
}

File: files/liskov/ParentChild.java
class Szulo {
    public void print(String s) {
        System.out.println(s);
    }
}

class Gyermek extends Szulo {
    public void echo(String s) {
        System.out.println(s);
        System.out.println(s);
    }
}

public class ParentChild {
    public static void main(String[] args) {
        Szulo sz = new Szulo();
        Szulo sz2 = new Gyermek();

        // Ez az, ami nem megy
        sz2.echo("ballalalalalala");
    }
}
